libs/cardano-ledger-binary/src/Cardano/Ledger/Binary/Group.hs

```diff
@@ -38,13 +40,10 @@ groupRecord = decodeRecordNamed "CBORGroup" (fromIntegral . toInteger . listLen)
 class EncCBORGroup a where
   encCBORGroup :: a -> Encoding
 
-  listLen :: a -> Word
+  listLen :: Proxy a -> Word
 
-  -- | an upper bound for 'listLen', used in 'Size' expressions.
-  listLenBound :: Proxy a -> Word
-
-listLenInt :: EncCBORGroup a => a -> Int
-listLenInt x = fromIntegral (listLen x)
+listLenInt :: forall a. EncCBORGroup a => a -> Int
```

Comment:
Why not a Proxy here too, instead of a value `a`? Isn't it misleading like this, as if the value is taken into consideration?